common/persistence/visibility/visibility_manager_dual.go

```diff
@@ -14,6 +16,8 @@ type (
 		secondaryVisibilityManager manager.VisibilityManager
 		managerSelector            managerSelector
 		enableShadowReadMode       dynamicconfig.BoolPropertyFn
+		shadowReadCtx              context.Context
```

Comment:
I'm honestly shocked that the linter passed. I think this is a strong signal we should improve this pattern

Reply:
Discussed offline